torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -359,6 +359,34 @@ at::Tensor AtenXlaType::_copy_from(const at::Tensor& self,
   return dst;
 }
 
+at::Tensor AtenXlaType::_copy_from_and_resize(const at::Tensor& self,
+                                              const at::Tensor& dst) {
+  XLA_FN_COUNTER("xla::");
+  auto dst_tensor = bridge::TryGetXlaTensor(dst);
+  auto self_tensor = bridge::TryGetXlaTensor(self);
+  if (!self_tensor) {
+    XLA_CHECK(dst_tensor);
+    dst_tensor->UpdateFromTensorOut(self);
```

Comment:
qq, in here you chose to call `UpdateFromTensorOut` which also handle the view check and defualt `sync-update` to false. Is there a reaon to do this instead of `dst_tensor->UpdateFromTensor(self, /*sync=*/sync_update);` in `AtenXlaType::_copy_from`?

Reply:
Yeah, that was actually my original motivation for writing this op. When I call `dst_tensor->UpdateFromTensor(self, /*sync=*/sync_update);` instead (like in _copy_from), xla complains if `dst` and `self` have different sizes. The next thing that I tried was to explicitly resize the dest tensor before the call, but then I hit https://github.com/pytorch/xla/issues/2881. I got the idea to call `UpdateFromTensorOut` from the `bridge::` API that the codegen used previously, which is [here](https://github.com/pytorch/xla/blob/7b538aacc9c9620b7533e89ccfbdbfef8e46fad1/torch_xla/csrc/aten_xla_bridge.cpp#L169). The only reason I didn't pass in `sync_update` was because `UpdateFromTensorOut` doesn't accept that arg :) but the original codegen also didn't use that argument, so I figured that this would be more in line with existing functionality.

Side note: implicitly resizing output tensors is actually currently allowed for in-tree kernels, but it's deprecated. So right now we want to allow that case, but eventually it'll probably go away. That's out of the scope of this function though, since that can be fixed through the codegen (we can just call _copy_from instead of _copy_from_and_resize).

Let me know if that all clears things up.

Reply:
Thanks for the explanation!